### PR TITLE
Feature/cypress uses mock login

### PR DESCRIPTION
**Pull request by pbastia**
Feature to bypass the login/hit to keycloak when running the e2e tests.
This reuses the same codepath as our `AS_ADMIN`, `AS_REPORTER`, ... targets and allows cypress to set a cookie to tell the server which user to take.

`app/server/postgraphile/authenticationPgSettings.js` checks for the AS_CYPRESS process argument, and a `mocks.auth` cookie to allow login to be bypassed as a certain test user.

### CODE REVIEW

File: app/cypress/integration/accessibility/admin-all-pages-accessibility.example.spec.js [hunk 1]
```diff
@@ -4,10 +4,7 @@ describe('When logged in as an admin', () => {
...
     // The admin spec uses the same setup as the analyst
     cy.sqlFixture('fixtures/analyst-all-access-setup');
-    cy.login(
-      Cypress.env('TEST_ADMIN_USERNAME'),
-      Cypress.env('TEST_ADMIN_PASSWORD')
-    );
+    cy.mockLogin('admin');
   });
 
   afterEach(() => {
```
Comment (matthieu-foucault) [inline app/cypress/integration/accessibility/admin-all-pages-accessibility.example.spec.js:10]: The `logout` below should be removed?

File: app/server/postgraphile/authenticationPgSettings.js [hunk 2]
```diff
@@ -17,11 +23,11 @@ const authenticationPgSettings = (req) => {
       'jwt.claims.sub': '00000000-0000-0000-0000-000000000000',
       'jwt.claims.user_groups': groups.join(','),
       'jwt.claims.priority_group': priorityGroup,
-      role: NO_AUTH_POSTGRES_ROLE
+      role: 'ciip_guest'
```
Comment (matthieu-foucault) [inline app/server/postgraphile/authenticationPgSettings.js:26]: Why this change? The NO_AUTH flag is intended for developers, and should be using whatever postgres superuser they have in their environment

File: app/cypress/integration/page-access.spec.js [hunk 3]
```diff
@@ -2,10 +2,7 @@ describe('When logged in as a reporter', () => {
   beforeEach(() => {
     cy.cleanSchema();
     cy.sqlFixture('dev/user');
-    cy.login(
-      Cypress.env('TEST_REPORTER_USERNAME'),
-      Cypress.env('TEST_REPORTER_PASSWORD')
-    );
+    cy.mockLogin('reporter');
```
Comment (matthieu-foucault) [inline app/cypress/integration/page-access.spec.js:5]: This file can be the one testing the real login

File: app/server/postgraphile/authenticationPgSettings.js (hunk 2 above)
Comment (pbastia) [reply app/server/postgraphile/authenticationPgSettings.js:26]: Oooh I completely misunderstood what this did, I missed it when I refactored this method out of index.js. Fix incoming

**Review by matthieu-foucault — APPROVED** [review]
:heart_eyes: 
Before: 
> All specs passed!                        28:24

After: 
> All specs passed!                    14:47

**Comment by kriscooke** [comment]
✨ 🌈
